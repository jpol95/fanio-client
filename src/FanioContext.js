import React from 'react'

const FanioContext = React.createContext({
    fandomList: [],
    reviewList: [],
    sectionList: [], 
    subList: [],
    tagList: [],
    reviewTagList: [],
    installmentList: [],
    handleSubmitReview: () => {}, 
    handleAddFandom: () => {}, 
    handleSubmitInstallments: () => {}, 
    handleGetReview: () => {}
})

export default FanioContext