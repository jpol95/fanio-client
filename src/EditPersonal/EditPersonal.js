import React from "react";
import "./EditPersonal.css";
import FetchService from '../FetchService'
import SignUpForm from '../SignUpForm/SignUpForm'
import FanioContext from '../FanioContext'

export default class EditPersonal extends React.Component {

  static contextType = FanioContext
  userId = Number(this.props.match.params.userId)

  handleSubmitUser = (e, personalInfo) => {
    e.preventDefault()
    const user = {
      fullname: personalInfo.fullname.value, 
      interests: personalInfo.interests.value.split(","),
      city: personalInfo.city.value, 
      education: personalInfo.education.value, 
      id: this.userId
    }
    FetchService.patchUser(user)
    .then((user) => {
      this.props.history.push(`/users/${this.userId}/profile`)
      this.context.handleEditPersonal(user)
    })
  }


  render() {
      const user = this.context.currentLoadedUser
      if (!user) return null
    return (

        <SignUpForm {...user} handleSubmitUser={this.handleSubmitUser} />
     
    );
  }
}
