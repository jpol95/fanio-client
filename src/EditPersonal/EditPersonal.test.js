import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import EditPersonal from './EditPersonal';
import FanioContext from '../FanioContext'

describe('Edit personal component', () => {
    const props = {
    match: {
        params: {
            userId: 1, 
            fandomId: 1, 
            installmentId: 1
        }
    }, 
    installId: 1
}
const currentLoadedUser = {
    id: 1,
    username: "kingbumii",
    fullname: "Jesse A Pollack",
    password: "$2y$12$4R1JkopQ4LgjXH27bUAV5OwezOQLoBP6Yv7mbd.Nv7V67yBSmepZq",
    education: "Purple University",
    interests: ["skating", "softball", "listending to show tunes", "knitting"],
    city: "Gallifrey",
  }
  

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FanioContext.Provider value={{currentLoadedUser}}> <EditPersonal {...props} /> </FanioContext.Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<FanioContext.Provider value={{currentLoadedUser}}> <EditPersonal {...props} /> </FanioContext.Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});