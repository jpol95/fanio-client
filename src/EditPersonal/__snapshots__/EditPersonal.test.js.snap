// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit personal component renders the UI as expected 1`] = `
Array [
  " ",
  <form
    className="new-user-form"
    onSubmit={[Function]}
  >
    <div
      className="personal"
    >
      <p>
        Tell us about yourself
      </p>
      <label
        htmlFor="fullname"
      >
        Full Name
      </label>
      <input
        defaultValue="Jesse A Pollack"
        id="fullname"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="interests"
      >
        Interests(comma separated list)
      </label>
      <input
        defaultValue={
          Array [
            "skating",
            "softball",
            "listending to show tunes",
            "knitting",
          ]
        }
        id="interests"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="city"
      >
        City
      </label>
      <input
        defaultValue="Gallifrey"
        id="city"
        onChange={[Function]}
        type="text"
      />
      <label
        htmlFor="education"
      >
        Education
      </label>
      <input
        defaultValue="Purple University"
        id="education"
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      className="submit-account-button"
      type="submit"
    >
      Submit
    </button>
  </form>,
  " ",
]
`;
